src/textual/app.py

```diff
@@ -566,22 +567,32 @@ def export_screenshot(self, *, title: str | None = None) -> str:
         console.print(screen_render)
         return console.export_svg(title=title or self.title)
 
-    def save_screenshot(self, path: str | None = None) -> str:
-        """Save a screenshot of the current screen.
+    def save_screenshot(
+        self,
+        filename: str | None = None,
+        path: str = "./",
+        time_format: str = "%Y-%m-%d %X %f",
+    ) -> str:
+        """Save a SVG screenshot of the current screen.
 
         Args:
-            path (str | None, optional): Path to SVG to save or None to pick
-                a filename automatically. Defaults to None.
+            filename (str | None, optional): Filename of SVG screenshot, or None to auto-generate
+                a filename with the data and time. Defaults to None.
+            path (str | None, optional): Path to directory for output or None for current
+                working directory. Default's to None"
+            time_format(str, optional): Time format to use if filename is None. Defaults to "%Y-%m-%d %X %f".
 
         Returns:
             str: Filename of screenshot.
         """
-        self.bell()
-        if path is None:
-            svg_path = f"{self.title.lower()}_{datetime.now().isoformat()}.svg"
-            svg_path = svg_path.replace("/", "_").replace("\\", "_")
+        if filename is None:
+            svg_filename = (
+                f"{self.title.lower()} {datetime.now().strftime(time_format)}.svg"
+            )
+            svg_filename = svg_filename.replace("/", "_").replace("\\", "_")
         else:
-            svg_path = path
+            svg_filename = filename
+        svg_path = os.path.expanduser(os.path.join(path, svg_filename))
```

Comment:
I suspect the comment above is just an out of date docstring, but if it ends up that `path` can be `None`, we need to make sure that it's handled here.